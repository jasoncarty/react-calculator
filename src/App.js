import React from "react";
import cs from "classnames";

import InputBlock from "./components/InputBlock";
import AnswerBlock from "./components/AnswerBlock";

import styles from "./App.module.css";

const App = () => {
  return (
    <div className={styles.app}>
      <div className={cs(styles.row)}>
        <InputBlock index={0} className={styles.block}></InputBlock>
        <InputBlock index={1} className={styles.block}></InputBlock>
        <InputBlock index={2} className={styles.block}></InputBlock>
      </div>
      <div className={cs(styles.row)}>
        <AnswerBlock className={styles.block}/>
      </div>
    </div>
  );
};

export default App;
