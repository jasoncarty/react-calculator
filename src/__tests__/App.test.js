import React from "react";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import ReactTestRenderer from 'react-test-renderer';

import App from "../App";

const mockStore = {
  subscribe: () => {},
  dispatch: () => {},
  getState: () => ({
    fieldValues: {},
    calculationType: ""
  })
};

describe("<App />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Provider store={mockStore}>
        <App />
      </Provider>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("matches snapshot", () => {
    const instance = ReactTestRenderer.create(
      <Provider store={mockStore}>
        <App />
      </Provider>
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });
});
