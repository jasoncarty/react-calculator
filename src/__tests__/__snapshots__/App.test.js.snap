// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="app"
>
  <div
    className="row"
  >
    <div
      className="inputBlock block"
    >
      <div
        className="content"
      >
        <div
          className="text"
        >
          Value 1: 
        </div>
        <input
          className="input"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="inputBlock block"
    >
      <div
        className="content"
      >
        <div
          className="text"
        >
          Value 2: 
        </div>
        <input
          className="input"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="inputBlock block"
    >
      <div
        className="content"
      >
        <div
          className="text"
        >
          Value 3: 
        </div>
        <input
          className="input"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="answerBlock block noMargin"
    >
      <div
        className="content"
      >
        <div
          className="radioList"
        >
          <div
            className="radioButton"
          >
            <input
              checked={false}
              name="calculationType"
              onChange={[Function]}
              type="radio"
              value="SUM"
            />
            <div
              className="text"
            >
              Sum
            </div>
          </div>
          <div
            className="radioButton"
          >
            <input
              checked={false}
              name="calculationType"
              onChange={[Function]}
              type="radio"
              value="MULTIPLY"
            />
            <div
              className="text"
            >
              Multiply
            </div>
          </div>
        </div>
        <div
          className="answer"
        >
          Result: 
          0
        </div>
      </div>
    </div>
  </div>
</div>
`;
